src/passes/Inlining.cpp

```diff
@@ -84,7 +84,7 @@ struct FunctionInfo {
     // about size, and if it's lightweight so a good candidate for
     // speeding us up.
     return options.optimizeLevel >= 3 && options.shrinkLevel == 0 &&
-           lightweight;
+           (lightweight || options.inlining.allowHeavyweight);
```

Comment:
Or better generalize this for any level?
```c++
return options.inlining.allowHeavyweight || 
       (options.optimizeLevel >= 3 && options.shrinkLevel == 0 && lightweight);
```

Reply:
I think it's simpler and better as it is.